system/jlib/jsocket.cpp

```diff
@@ -4914,23 +4917,33 @@ class CSocketEpollThread: public CSocketBaseThread
         if (removeSock(sock))
         {
             selectvarschange = true;
+            // NB: could set terminating here if no more hdls on
+            // this thread and at least one other thread is present
             triggerselect();
             return true;
         }
         return false;
     }
 
-    bool add(ISocket *sock,unsigned mode,ISocketSelectNotify *nfy)
+    unsigned add(ISocket *sock,unsigned mode,ISocketSelectNotify *nfy)
     {
         if ( !sock || !nfy ||
              !(mode & (SELECTMODE_READ|SELECTMODE_WRITE|SELECTMODE_EXCEPT)) )
         {
             WARNLOG("EPOLL: adding fd but sock or nfy is NULL or mode is empty");
             dbgassertex(false);
-            return false;
+            return 0;
         }
         CriticalBlock block(sect);
-        removeSock(sock);
+        if (terminating)
+            return 0;
+        unsigned rm = 0;
+        if (removeSock(sock))
+            rm = 0x2;
```

Comment:
This really should use flags from an enumeration or similar - otherwise it isn't clear what is being checked.